common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java

```diff
@@ -0,0 +1,159 @@
+        }
+        return msRequest;
+    }
+
+    @NonNull
+    private static MicrosoftStsOAuth2Strategy asMicrosoftStsOAuth2Strategy(final OAuth2Strategy strategy) {
+        MicrosoftStsOAuth2Strategy msStrategy;
+        if (strategy instanceof MicrosoftStsOAuth2Strategy) {
+            msStrategy = (MicrosoftStsOAuth2Strategy) strategy;
+        } else {
+            throw new IllegalArgumentException("Invalid strategy type.");
+        }
+        return msStrategy;
+    }
+
+    @Override
+    public AccessToken createAccessToken(
+            final OAuth2Strategy strategy,
+            final AuthorizationRequest request,
+            final TokenResponse response) {
+        final AccessToken accessToken = new AccessToken();
+        accessToken.setTarget(getTarget(request));
+        final long cachedAt = getCachedAt();
+        accessToken.setCachedAt(String.valueOf(cachedAt)); // generated @ client side
+        final long expiresOn = getExpiresOn(cachedAt, response);
+        accessToken.setExpiresOn(String.valueOf(expiresOn)); // derived from expires_in
+        accessToken.setClientInfo(getClientInfo(response));
+        // TODO Do AccessTokens track a family id?
+        //accessToken.setFamilyId(msTokenResponse.getFamilyId());
+
+        return accessToken;
+    }
+
+    @Override
+    public RefreshToken createRefreshToken(
+            final OAuth2Strategy strategy,
+            final AuthorizationRequest request,
+            final TokenResponse response) {
+        final RefreshToken refreshToken = new RefreshToken();
+        refreshToken.setTarget(getTarget(request));
+        final long cachedAt = getCachedAt();
+        refreshToken.setCachedAt(String.valueOf(cachedAt)); // generated @ client side
+        /*
+        Per this document: https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-protocols-oauth-code
+        expires_on is expressed as SECONDS since Jan 1 1970 - our schema caches this value as millis derived from expires_in
+         */
+        final long expiresOn = getExpiresOn(cachedAt, response);
+        refreshToken.setExpiresOn(String.valueOf(expiresOn)); // derived from expires_in
+        refreshToken.setClientInfo(getClientInfo(response));
+        refreshToken.setFamilyId(getFamilyId(response));
+        return refreshToken;
+    }
+
+    private String getTarget(final AuthorizationRequest request) {
+        final MicrosoftStsAuthorizationRequest msRequest = asMicrosoftStsAuthorizationRequest(request);
+        return msRequest.getScope();
+    }
+
+    private long getCachedAt() {
+        return System.currentTimeMillis();
+    }
+
+    private long getExpiresOn(final long cachedAt, final TokenResponse response) {
+        final MicrosoftStsTokenResponse msTokenResponse = asMicrosoftStsTokenResponse(response);
+        final long expiresInSeconds = msTokenResponse.getExpiresIn();
+        final long expiresInMillis = expiresInSeconds * 1000L;
```

Comment:
This value is now going to be seconds -- will update in a future commit

/cc @shoatman